src/pages/iou/ReceiptSelector/index.native.js

```diff
@@ -0,0 +1,313 @@
+                    onPress: () => Linking.openSettings(),
+                },
+            ],
+            {cancelable: false},
+        );
+    };
+
+    /**
+     * A generic handling when we don't know the exact reason for an error
+     *
+     */
+    const showGeneralAlert = () => {
+        Alert.alert(translate('attachmentPicker.attachmentError'), translate('attachmentPicker.errorWhileSelectingAttachment'));
+    };
+
+    const showCameraAlert = () => {
+        Alert.alert(translate('receipt.cameraErrorTitle'), translate('receipt.cameraErrorMessage'));
+    };
+
+    /**
+     * Common image picker handling
+     *
+     * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
+     * @returns {Promise<ImagePickerResponse>}
+     */
+    const showImagePicker = (imagePickerFunc) =>
+        new Promise((resolve, reject) => {
+            imagePickerFunc(getImagePickerOptions(CONST.ATTACHMENT_PICKER_TYPE.IMAGE), (response) => {
+                if (response.didCancel) {
+                    // When the user cancelled resolve with no attachment
+                    return resolve();
+                }
+                if (response.errorCode) {
+                    switch (response.errorCode) {
+                        case 'permission':
+                            showPermissionsAlert();
+                            return resolve();
+                        default:
+                            showGeneralAlert();
+                            break;
+                    }
+
+                    return reject(new Error(`Error during attachment selection: ${response.errorMessage}`));
+                }
+
+                return resolve(response.assets);
+            });
+        });
+
+    const takePhoto = () => {
+        if (!camera.current) {
+            showCameraAlert();
+            return;
+        }
+
+        camera.current
+            .takePhoto({
+                qualityPrioritization: 'speed',
+                flash: flash ? 'on' : 'off',
+            })
+            .then((photo) => {
+                IOU.setMoneyRequestReceipt(`file://${photo.path}`, photo.path);
+                IOU.navigateToNextPage(props.iou, reportID.current, props.report);
+            })
+            .catch(() => {
+                showCameraAlert();
+            });
+    };
+
+    Camera.getCameraPermissionStatus().then((permissionStatus) => {
+        setPermissions(permissionStatus);
+    });
+
+    const cameraComponent = () => (
+        <Camera
+            ref={camera}
+            device={device}
+            style={[styles.cameraView]}
+            isActive
+            photo
+        />
+    );
+
+    const loadingComponent = () => (
+        <View style={[styles.cameraView]}>
+            <ActivityIndicator
+                size="large"
+                style={{flex: 1}}
+                color={themeColors.textSupporting}
+            />
+        </View>
+    );
```

Comment:
We don't need to define those. Just render them when needed both are only referenced once.